C7GameData/SaveFormat.cs

```diff
@@ -10,27 +10,59 @@ namespace C7GameData
 */
 {
     using System.IO;
+    using System.IO.Compression;
     using System.Text.Json;
+
+    public enum SaveFileFormat
+    {
+        Plain,
+        Zip,
+    }
+
     public class C7SaveFormat
     {
         public string Version = "v0.0early-prototype";
+
         // Rules is intended to be the analog to a BIC/X/Q
         public C7RulesFormat Rules;
+
         // This naming is probably bad form, but it makes sense to me to name it as such here
         public GameData GameData;
+
         public C7SaveFormat()
         {
 	        GameData = new GameData();
         }
+
         public C7SaveFormat(GameData gameData, C7RulesFormat rules = null)
         {
-            this.GameData = gameData;
+            GameData = gameData;
             Rules = rules;
         }
-        public static C7SaveFormat Load(string path)
+
+        public static C7SaveFormat Load(string path, SaveFileFormat format = SaveFileFormat.Plain)
         {
-            string json = File.ReadAllText(path);
-            C7SaveFormat save = JsonSerializer.Deserialize<C7SaveFormat>(json, JsonOptions);
+            System.Console.WriteLine("loading...");
+            C7SaveFormat save = null;
+            if (format == SaveFileFormat.Plain)
+            {
+                string json = File.ReadAllText(path);
+                save = JsonSerializer.Deserialize<C7SaveFormat>(json, JsonOptions);
+            }
+            else
+            {
+                using (var fileStream = new FileStream(path, FileMode.Open))
+                using (var archive = new ZipArchive(fileStream, ZipArchiveMode.Read))
+                {
+                    ZipArchiveEntry entry = archive.GetEntry("save");
+                    using (Stream stream = entry.Open())
+                    {
+                        var task = JsonSerializer.DeserializeAsync<C7SaveFormat>(stream, JsonOptions).AsTask();
+                        save = task.GetAwaiter().GetResult();
```

Comment:
Is there any practical benefit to doing this async?  As I read it, the thread that spins off the async task on line 60 is just going to immediately pause on line 61 to wait for the result.  So I don't think there's any benefit to it being async?

(Disclaimer: Not a C# expert yet.  So maybe there is?)

Reply:
You're right in that there is no performance benefit since 61 blocks until the result is available. However, since the zip archive only gives a stream, this seemed better than reading from the stream into a buffer in memory and then doing the deserialisation on the buffer (and maybe this uses slightly less memory for really large saves? I'm also new to C#)

Reply:
I was thinking just the following:

```
var task = JsonSerializer.Deserialize<C7SaveFormat>(stream, JsonOptions);
```

They seem to be equivalent behind the scenes; from the parameter name I suspect it's a stream in both cases.

Good to know you're also new to C#!  For some reason I thought you were a C# developer; maybe we should start a list of who our C# experts are.

Reply:
The interface is pretty bad, but in order for GetResult to block until the job is actually finished, the `AsTask()` is required, if my understanding of [this documentation](https://devblogs.microsoft.com/dotnet/understanding-the-whys-whats-and-whens-of-valuetask/#valid-consumption-patterns-for-valuetasks) is correct